extensions/amp-ad/0.1/amp-ad-ui.js

```diff
@@ -198,6 +201,49 @@ export class AmpAdUIHandler {
     this.baseInstance_.element.appendChild(uiComponent);
     return uiComponent;
   }
+
+  /**
+   * @param {number|string|undefined} height
+   * @param {number|string|undefined} width
+   * @param {number} iframeHeight
+   * @param {number} iframeWidth
+   * @return {!Promise<Object>}
+   */
+  updateSize(height, width, iframeHeight, iframeWidth) {
+    // Calculate new width and height of the container to include the padding.
+    // If padding is negative, just use the requested width and height directly.
+    let newHeight, newWidth;
+    height = parseInt(height, 10);
+    if (!isNaN(height)) {
+      newHeight = Math.max(this.element_./*OK*/offsetHeight +
+          height - iframeHeight, height);
+    }
+    width = parseInt(width, 10);
+    if (!isNaN(width)) {
+      newWidth = Math.max(this.element_./*OK*/offsetWidth +
+          width - iframeWidth, width);
+    }
+
+    const sizes = {
+      newWidth,
+      newHeight,
+    };
+
+    if (!newHeight && !newWidth) {
+      return Promise.reject(sizes);
+    }
+
+    if (getAdContainer(this.element_) == 'AMP-STICKY-AD') {
+      // Special case: force collapse sticky-ad if no content.
+      return Promise.reject(sizes);
+    }
+    return this.baseInstance_.attemptChangeSize(
+        newHeight, newWidth).then(() => {
+          return Promise.resolve(sizes);
+        }, () => {
+          return Promise.reject(sizes);
```

Comment:
Ah! Promise has to reject with an error. But I really need to pass the sizes object back. In this case I will probably always resolve with a param to indicate the success of attemptChangeSize. What do you think?
BTW  I do see few usage like https://github.com/ampproject/amphtml/blob/master/extensions/amp-fx-flying-carpet/0.1/test/test-amp-fx-flying-carpet.js#L72. And the current reject with sizes object works. Quite surprised.